vaadin-list-box-flow-parent/vaadin-list-box-flow/src/main/java/com/vaadin/flow/component/listbox/ListBoxBase.java

```diff
@@ -102,6 +102,7 @@ public void setDataProvider(DataProvider<ITEM, ?> dataProvider) {
         this.dataProvider.set(Objects.requireNonNull(dataProvider));
         DataViewUtils.removeComponentFilterAndSortComparator(this);
         clear();
+        rebuild();
```

Comment:
Did you just fix another regression or what is this change about ? 😄 

Wondering if it should be highlighted as an additional fix. Was there a test failure that occurred that caught it ? IT? Or unit test ?

Reply:
Good question!
This was another regression, which was revealed when I deleted registration `null` check in `ListBox`.
It fails `ListBoxIT.componentsBetweenItems` tests and the reason was that `rebuild()` method was in the wrong place and it was called mistakenly in `setupDataProviderListener`.
I placed it next to `clear()` method, as it done in the data provider listener.

I can place it into additional fix, but then I'll need to revert all changes in ListBox in this PR.

Reply:
It is fine I think (like this), but please just highlight this in the commit description when merging.